### DISCUSSION TITLE
Transcribe User Interface

### DISCUSSION THREAD

**Discussion by vivekuppal** [vivekuppal/transcribe#121] [General]
We have received feedback on the User Interface and attempted to make it better by going towards a browser based UI.
This is fairly major undertaking and complete revamp.
Good news is that it does not require much change to many of the classes we have already in place. The new UI can co-exist alongside the existing UI at least thus far :-) 

Goals for UI overhaul are fairly simple

1. Modern UI, easier to maintain than the current UI
2. Easier UX from end user perspective
3. Allow change of settings without restart and without changing text files

Sharing some screenshots here to provide an opportunity for feedback.

This is very early days of the UI, and we can go in any direction. All feedback welcome.

![image](https://github.com/vivekuppal/transcribe/assets/11325412/05d4aef9-eb3d-4b02-be2c-b01ddbde6131)

![image](https://github.com/vivekuppal/transcribe/assets/11325412/4cbc23d1-1a3b-4041-b5da-1e9433f0e818)

**Comment by j0n0w1ns** [comment]
Does the new UI support all the functionality available in the existing UI?
How can we get access to the new UI to experiment with it?

**Comment by mang0sw33t** [comment]
What tech stack are you using to build the UI?
Want to get a way to experiment with it.

**Comment by vivekuppal** [comment]
@j0n0w1ns 
>Does the new UI support all the functionality available in the existing UI?

Ultimate goal is to support all functionality. At the moment we have prioritized features most requested in feedback. E.g. Ability to change Mic, Speakers in the UI without restarting the application. Change some of the configuration parameters without changing text files and such.

Prioritized feedback on features you would like to see in the UI would be extremely helpful

> How can we get access to the new UI to experiment with it?

Working on a way to do that

@mang0sw33t 
>What tech stack are you using to build the UI?

dash / plotly - https://plotly.com/dash/ 

> Want to get a way to experiment with it.

Working on that

**Comment by Adarsha-gg** [comment]
The main page looks fine but the settings page looks kind of elongated (bigger texts heights maybe?) and I think placing the two settings box side by side would be a better look as it reduces white space.
We could also make the Api- key as password box (••••) as mostly api keys are copy pasted anyways and for privacy.

**Comment by victoryg739** [comment]
Does the new web version work on macOS?

**Reply by vivekuppal** [reply]
It is for windows
